### PR TITLE
Wip rocksdb conf

### PR DESCRIPTION
**Pull request by xiaoxichen**
We would like to use GetOptionsFromString

Now some of the conf are not supported, will interpret them:
    1. compact on mount
    2. disableWAL
    3. threadnum

### CODE REVIEW

File: src/common/config_opts.h [hunk 1]
```diff
@@ -701,39 +701,12 @@ OPTION(kinetic_hmac_key, OPT_STR, "asdfasdf") // kinetic key to authenticate wit
...
-OPTION(rocksdb_log, OPT_STR, "/dev/null")  // enable rocksdb log file
-OPTION(rocksdb_info_log_level, OPT_STR, "info")  // info log level : debug , info , warn, error, fatal
-OPTION(rocksdb_wal_dir, OPT_STR, "")  //  rocksdb write ahead log file, put it to fast device will benifit wrtie performance
-OPTION(rocksdb_disableDataSync, OPT_BOOL, false) // if true, data files are not synced to stable storage
-OPTION(rocksdb_disableWAL, OPT_BOOL, false)  // if true, writes will not first go to the write ahead log
-
+// rocksdb options that will be used for keyvaluestore(if backend is rocksdb)
+OPTION(keyvaluestore_rocksdb_options, OPT_STR, "")
+// rocksdb options that will be used for omap(if omap_backend is rocksdb)
+OPTION(omap_rocksdb_options, OPT_STR, "")
```
Comment (liewegas) [inline src/common/config_opts.h]: filestore_rocksdb_options?

**Comment by liewegas** [comment]
rest looks good!

**Comment by xiaoxichen** [comment]
@liewegas ,I think the work is done here, see if it looks good for you...but @loic-bot is offline now..

**Comment by xiaoxichen** [comment]
@dachary , I suspect the robot just jump in accidently, this build is not for me:)

**Comment by tchaikov** [comment]
@xiaoxichen ,  the build of http://jenkins.ceph.dachary.org/job/ceph/5157/ grabbed the commit of 5895699fb13e96e3b3c3b73c67a5f85e8493c92f, i.e. your last commit of "test/KeyValueDBMemory : update init()", but ignored the others.

**Comment by xiaoxichen** [comment]
@liewegas ,fixed and seems bot is happy

**Pull request merged**